RIPS/rip-9998.md

```diff
@@ -0,0 +1,306 @@
+discussions-to: https://www.google.com/
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-09-01
+requires: 7560
+---
+
+## Abstract
+
+RIP-7560 defined a native way to achieve Account Abstraction on Ethereum. However, one big limitation remains:
+each transaction must carry its own `signature` or other form of validation input in order to be included.
+
+We propose an extension to the RIP-7560 that introduces a validation frame that is shared by multiple transactions.
+
+This addition follows the design of [ERC-4337](./eip-4337) signature aggregation and will enable
+Ethereum transactions to natively support sharing validation inputs for the first time.
+
+## Motivation
+
+Using validation schemes that allow signature aggregation enables significant optimisations and savings on
+gas for execution and transaction data cost. This is especially relevant in the context of rollups that publish data on
+the Ethereum mainnet.
+
+Another reason to support aggregation for transactions is snark aggregation.
+Some signature schemes, such as the ones relying on `secp256r1`, are expensive to verify.
+An aggregated zero-knowledge proof that the signatures of a large number of transactions were verified
+may be cheaper to verify on-chain than verifying each of them separately.
+
+## Specification
+
+### Transaction version AA_TX_TYPE_AGGREGATED_SIG with aggregated signature data
+
+A new special version of RIP-7560 Transaction called `AA_TX_TYPE_AGGREGATED_SIG` is introduced to carry
+the aggregated signature data:
+
+```
+
+0x04 || 0x01 || rlp([chainId, transactionCount, aggregator, aggregatedSignature, aggregatedSigValidationGasLimit, accessList])
+
+```
+
+All on-chain computations that take place during the aggregated signature validation happen within the context
+of this transaction.
+
+### Calculation of Transaction Type AA_TX_TYPE_AGGREGATED_SIG hash
+
+```
+
+keccak256(0x04 || 0x01 || rlp(transaction_payload)
+
+```
+
+### Bundles of aggregated transactions
+
+As aggregated transactions share their signatures, they form a set of transactions that are only valid together and in
+a given order. We refer to this set of transactions as "aggregated bundle".
+
+Upon inclusion in a block, each bundle consists of the single "Aggregator validation transaction" and an arbitrary
+number of individual aggregated AA transactions.
+
+There may be an arbitrary number of bundles in an Ethereum block, using same or different `aggregator` addresses.
+
+### Aggregator validation transaction
+
+The aggregator validation transaction must run before any of the transactions authorized by this `aggregator`.
+
+If the execution of this transaction results in a revert the block is INVALID.
+
+We define the following Solidity method and the `aggregator` of the transaction is invoked with a corresponding data:
+
+```solidity
+
+function validateSignatures(uint256 version, bytes[] aggregatedTransactions, bytes aggregatedSignature) external view;
```

Comment:
this is a kind of method that is hard to be replaced with EOF section.
I has 2-d input (array if bytes), which requires opionated encoding.
OTOH, there is less chance of "signature collision", since it is only in the aggregator itself, not account or paymaster.

Reply:
Yes, encoding is needed.  See our current discussion re SSZ.  We still don't need a method for it.  Just another EOF "entrypoint" which will be called with an array of SSZ-encoded transactions and the aggregated sig.